rapi sender class test

sandeen opened the pull request:
Here's a first stab at using @lincomatic 's RapiSender class.
It has some slight modifications, first to make the coding style consistent w/ this project, and a small change to allow a request of the original response string for the RAPI tab.  (I'll probably revert that and just iterate over the tokens in the response...)

Note that this puts the serial delay in the main loop, but playing with it a little, it seems ok.  See what you think.

It could also use some smarts to decide when it's OK to turn on the sequence ID feature.

Related to that, how far back is this project supposed to be compatible w/ OpenEVS version and/or protocol version?  It might need a little care in that respect as well.

Anyway, seems like a decent first cut, let me know what you think.

jeremypoulter commented on src/input.cpp:
```diff
@@ -98,246 +96,146 @@ create_rapi_json() {
+      String qrapi = rapiSender.getToken(1);
+      state = strtol(qrapi.c_str(), NULL, 16);
+      switch (state) {
+        case 1:
```
Can this be an #define or enum, 'magic' numbers are bad for code readability (same comment for every number here)

sandeen replied:
Yeah, that's just inherited, really, modulo whitespace in this patch.  That's been on my radar too, but is probably material for a different changeset.

jeremypoulter commented on src/wifi.cpp:
```diff
@@ -162,22 +156,14 @@ startClient() {
-    Serial.println(tmpStr);
     DEBUG.println(tmpStr);
+    rapiSender.sendCmd("$FP 0 0 Client-IP.......");
+    snprintf(tmpStr, 40, "%s %s", "$FP 0 1", ipaddress.c_str());
```
Why the first %s? just put the constant string there

sandeen replied:
Haha, good point.  No idea why I did that.  :)  derp!

jeremypoulter commented on src/RapiSender.cpp:
Consider putting this in a (inline) method as it is duplicated a few times and a well named method will also help readability

sandeen commented on src/RapiSender.cpp:
yeah, fair enough  - I haven't really gone through RapiSender.cpp closely, it's mostly just as sent because it worked as sent...  I'm sure there are a few things to clean up.  Thanks!

sandeen commented:
One thing to note is that the RAPI tab works quite nicely with this change ;)

glynhudson commented:
Fantastic 👍 . Looking forward to testing when I'm home next week.

>Related to that, how far back is this project supposed to be compatible w/ OpenEVS version and/or protocol version? It might need a little care in that respect as well.

This ^ is a question for @chris1howell . How many user running older versions of OpeneEVSE do you expect to be using the ESP WiFi? Shall we focus on supporting the latest versions of OpenEVSE controller? rapi firmware?

chris1howell commented:
The most common code out there is the latest "stable" release 3.11.3. I am working on moving my devolopment environment to platform IO and testing and pushing out a release from lincomatics development branch.

So the best baseline for today is 3.11.3.


On Thu, May 25, 2017 at 12:46 PM, Glyn Hudson <notifications@github.com<mailto:notifications@github.com>> wrote:

Fantastic 👍 . Looking forward to testing when I'm home next week.

Related to that, how far back is this project supposed to be compatible w/ OpenEVS version and/or protocol version? It might need a little care in that respect as well.

This ^ is a question for @chris1howell<https://github.com/chris1howell> . How many user running older versions of OpeneEVSE do you expect to be using the ESP WiFi? Shall we focus on supporting the latest versions of OpenEVSE controller? rapi firmware?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/OpenEVSE/ESP8266_WiFi_v2.x/pull/20#issuecomment-304105069>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ABvYH0RCOeDXfuptZCp_YL_7aC5JKnVuks5r9dp8gaJpZM4NmkdH>.

sandeen commented:
I'm new at this github thing ;) so not quite sure where to bring this up, but something changed recently in the tree... the conflict above shows it - `#define DEBUG_SERIAL1 `got commented out in src/emonesp.h, and now debug messages are going out on the main serial line.  The solar divert tab is also now visible by default.  It seems like a few unintentional changes got in, but I can't quite tell how it happened; git blame shows that the DEBUG_SERIAL1 change is from March.
home.html got renamed to home.htm, input.cpp gained new whitespace damage, etc.

jeremypoulter commented:
I would just raise them as issues and they can be fixed as appropriate. @glynhudson can probably comment better on the Solar PV changes, but it is highly likely that I have missed something or otherwise broken something due to the architecture changes, please do question anything you notice.

sandeen commented:
Ok, it seems more of a mystery mis-merge than localized bugs - renaming home.html for example doesn't look intentional.  I raised a blanket issue for the recent oddities, thanks.

sandeen commented on src/RapiSender.cpp:
Ok despite my previous comment I'm not quite sure what part you're suggesting should be an inline...

jeremypoulter commented on src/RapiSender.cpp:
The check sum calculations, lines 60-62 and 68-70

jeremypoulter commented:
@sandeen what are your plans with this PR? I really need to sort out the robustness of the RAPI comms for #34.

sandeen commented:
I've been crushed at Real Work (tm) lately, so haven't been hacking on this so much.  I can try to get it up to date again, and maybe the minor issues raised above can be fixed up as later commits?  (I fixed the dumb string one...)

jeremypoulter commented:
That is fine with me.

On Jun 8, 2017 12:26 AM, "Eric Sandeen" <notifications@github.com> wrote:

> I've been crushed at Real Work (tm) lately, so haven't been hacking on
> this so much. I can try to get it up to date again, and maybe the minor
> issues raised above can be fixed up as later commits? (I fixed the dumb
> string one...)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenEVSE/ESP8266_WiFi_v2.x/pull/20#issuecomment-306954213>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AF1WgAyGCmdXxdColUstH6lm9iInTh-7ks5sBzGngaJpZM4NmkdH>
> .
>

sandeen commented:
Ok, I've rebased this branch on top of the master branch.  I apologize - this rebase isn't tested, I'm just too swamped right now.  If you want to give it a whirl, here it is.

Thanks!
-Eric

jeremypoulter commented:
Looks to work after making one fix to work with no checksum: https://github.com/jeremypoulter/ESP8266_WiFi_v2.x/commit/d754e8eed004591932009f542dc5b2596f9dea36

@glynhudson I think this is good to merge.

sandeen commented:
Hm, ok - so, I rebased this branch a few days ago, but it was missing one local change which caused it to not compile.  :(  So I'm wondering what @jeremypoulter tested...
I've rebased it again with that fix.  (Sorry, maybe rebasing makes things a mess, I can push a fresh branch name if you like).
I'm getting everything loaded up to confirm versions, behavior, etc - Jeremy spooked me a little. ;)

jeremypoulter commented:
I merged on to the branch I was working on as I needed the changes to the
the /rapi endpoint working, maybe I fixed the issue you saw in the process
:( sorry for the confusion.

On 12 June 2017 at 22:40, Eric Sandeen <notifications@github.com> wrote:

> Hm, ok - so, I rebased this branch a few days ago, but it was missing one
> local change which caused it to not compile. :( So I'm wondering what
> @jeremypoulter <https://github.com/jeremypoulter> tested...
> I've rebased it again with that fix. (Sorry, maybe rebasing makes things a
> mess, I can push a fresh branch name if you like).
> I'm getting everything loaded up to confirm versions, behavior, etc -
> Jeremy spooked me a little. ;)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenEVSE/ESP8266_WiFi_v2.x/pull/20#issuecomment-307937354>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AF1WgA3_nocmbj1RaM2PQWencbgbsgbqks5sDbBbgaJpZM4NmkdH>
> .
>

sandeen commented:
Ok, I'm still having the same problems as I had earlier with the master branch - pages just don't load for me.  I'm not going to be able to dig in and try to debug that for a while I'm afraid.

Still - I pushed a rebased "rapisender" branch today fixing a build problem :/ so if you play with this, be sure to do a forced update from this branch.

sandeen commented:
(sorry for pushing something that didn't build, I use _guilt_ to manage a patch stack but if I forget to refresh the top patch before pushing to github, it misses local changes.  My bad)
